tests/network/persistentips.go

```diff
@@ -0,0 +1,158 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	ipamclaimsv1alpha1 "github.com/k8snetworkplumbingwg/ipamclaims/pkg/crd/ipamclaims/v1alpha1"
+	nadv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	kvirtv1 "kubevirt.io/api/core/v1"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/framework/checks"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = SIGDescribe("Persistent IPs", func() {
+
+	Context("with a VirtualMachine configured with a allowPersistentIPs nad", func() {
+		var (
+			vm         *kvirtv1.VirtualMachine
+			vmi        *kvirtv1.VirtualMachineInstance
+			nad        *nadv1.NetworkAttachmentDefinition
+			refreshVMI = func() error {
+				var err error
+				vmi, err = kubevirt.Client().VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+			currentVMIUD = func() (types.UID, error) {
+				if err := refreshVMI(); err != nil {
+					return "", err
+				}
+				return vmi.UID, nil
+			}
```

Comment:
usually we extract inline functions as part of the sig code-quality
please consider doing that (same for the others)

Reply:
This closures use test suite global state, they just improve readability at `Eventually` I don't think is worth it to extract them.

Reply:
there is a big effort all over the code base to remove those, and inline client get whenever needed,
it just matter of time until someone will refactor it imho

Reply:
Ack, I have move to a common place the stuff related to IPAMClaims and also use the matchers we have already.